R/data.table.R

```diff
@@ -1012,6 +1012,11 @@ replace_dot_alias = function(e) {
         # .SDcols might include grouping columns if users wants that, but normally we expect user not to include them in .SDcols
       } else {
         if (!missing(.SDcols)) warning("This j doesn't use .SD but .SDcols has been supplied. Ignoring .SDcols. See ?data.table.")
+        ## VIOLATES TESTED BEHAVIOR WRT DUPLICATES, SEE TESTS 1290 **
+        if (anyDuplicated(used_from_x <- dupintersect(names_x, av))) {
+          dupcols = unique(used_from_x[duplicated(used_from_x)])
+          stop("Unable to disambiguate reference to duplicated columns in x: ", brackify(dupcols))
```

Comment:
Maybe:
```r
stop(gettextf("Unable to disambiguate to duplicated columns in %s: %s", deparse(substitute(x)), brackify(dupcols), domain = ...)
```
This will allow the original name of the ```data.table``` to print instead of ```x```.

Reply:
This is great in simple cases, but e.g. `data.table(x = 1)[[ , c('y', 'y') := .(1, 2)]` and other dynamic cases, or when a function the user calls is the one that creates the error, it may not be as helpful. There might be an argument for building something like `guess_name(x)` that checks for the "non-standard" cases and returns `x`, otherwise a user-friendly name, but that also comes with some maintenance burden...

Reply:
That is a good point. Here is a simple alternative that should both be maintainable but still allow for user-friendly messages.  
``` r
fx = function(x) {
  print(if (is.name(x_sub <- substitute(x))) deparse(x_sub) else "x")
}

fx(iris)
#> [1] "iris"
fx(iris[])
#> [1] "x"
```